### DISCUSSION TITLE
Build failed on Expanse

### DISCUSSION THREAD

**Discussion by benjamin-cash** [dtcenter/METplus#1059] [Installation] [answered]
My attempt to build on Expanse failed with the following:
make[4]: Entering directory '/expanse/lustre/projects/mia328/bcash/metplus/MET/met/src/basic/vx_math'
make[4]: *** No rule to make target 'viewgravity_to_string.cc', needed by 'libvx_math_a-viewgravity_to_string.o'.  Stop.
make[4]: Leaving directory '/expanse/lustre/projects/mia328/bcash/metplus/MET/met/src/basic/vx_math'
make[3]: *** [Makefile:367: all-recursive] Error 1
make[3]: Leaving directory '/expanse/lustre/projects/mia328/bcash/metplus/MET/met/src/basic'
make[2]: *** [Makefile:369: all-recursive] Error 1
make[2]: Leaving directory '/expanse/lustre/projects/mia328/bcash/metplus/MET/met/src'
make[1]: *** [Makefile:424: all-recursive] Error 1
make[1]: Leaving directory '/expanse/lustre/projects/mia328/bcash/metplus/MET/met'
make: *** [Makefile:365: all] Error 2

Suggestions for where I went wrong?

Edit: Attempting to build on stampede2 fails at the same point.

**Comment by georgemccabe** [comment]
Hi @benjamin-cash , where did you obtain the MET source code? The source code tarfile should be obtained from the MET Downloads page: https://dtcenter.org/community-code/metplus/met-version-10-0-0

Here is the direct link for MET v10.0.0: https://github.com/dtcenter/MET/releases/download/v10.0.0/met-10.0.0.20210510.tar.gz

The tarfile that is automatically generated for a GitHub release contains additional files meant for development that are removed from the final tarfile that is linked on the downloads page. This is likely the cause for your error. Please try installing from the tarfile listed above and let me know if you run into any issues.

**Reply by benjamin-cash** [reply]
HI @georgemccabe , I obtained the code via: git clone https://github.com/dtcenter/metplus and then running build_components/build_MET.sh, following the directions here: https://metplus.readthedocs.io/en/main_v4.0/Users_Guide/installation.html. Should I not be using that procedure?

**Comment by georgemccabe** [comment, marked as answer]
Hi @benjamin-cash,

I apologize for the confusion. It appears that the scripts used to obtain and build METplus this way need to be updated. Thank you for bringing this to our attention. I will update the instructions on the METplus User's Guide.

I recommend navigating to the Downloads page for the MET software and follow the instructions under the "Sample Script for Compiling External Libraries and MET" heading.
https://dtcenter.org/community-code/model-evaluation-tools-met/download

You will need download the latest release from this page (10.0.0):
https://github.com/dtcenter/MET/releases/download/v10.0.0/met-10.0.0.20210510.tar.gz

Once the MET executables have been successfully installed, you can continue to the System Configuration section of the METplus User's Guide to configure the METplus wrappers to use the MET binaries that you have installed.
https://metplus.readthedocs.io/en/main_v4.0/Users_Guide/systemconfiguration.html#system-configuration

Please let me know if you run into any issues with the installation process.

Thanks,
@georgemccabe

**Reply by benjamin-cash** [reply]
Hi @georgemccabe - I tried following this procedure and I made it farther along but when I tried running the example I hit the following:

08/06 12:29:42.121 metplus (met_util.py:67) INFO: Running METplus v4.0.0 called with command: ./run_metplus.py -c /work/02441/bcash/stampede2/metplus/METplus-4.0.0/parm/use_cases/met_tool_wrapper/Example/Example.conf -c /work/02441/bcash/stampede2/metplus/METplus-4.0.0/parm/metplus_config/defaults.conf
08/06 12:29:42.121 metplus (met_util.py:70) INFO: Log file: /scratch/02441/bcash/metplus/logs/metplus.log.20210806122942
08/06 12:29:42.126 metplus.Usage (met_util.py:186) ERROR: Fatal error occurred
Traceback (most recent call last):
  File "/work2/02441/bcash/stampede2/metplus/METplus-4.0.0/metplus/util/met_util.py", line 127, in run_metplus
    getattr(module, f"{process}Wrapper")(config,
  File "/work2/02441/bcash/stampede2/metplus/METplus-4.0.0/metplus/wrappers/usage_wrapper.py", line 19, in __init__
    self.available_processes = list(set(val for val in met_util.LOWER_TO_WRAPPER_NAME.values()))
AttributeError: module 'metplus.util.met_util' has no attribute 'LOWER_TO_WRAPPER_NAME'

I was testing this on stampede2 so that I could use the existing settings and just make sure I could get through the build. That error message doesn't give me a lot to go on, unfortunately. Any thoughts?

**Reply by georgemccabe** [reply]
Hi @benjamin-cash, the error that occurs here is a minor bug in the UsageWrapper that has been fixed since 4.0.0 was released. However, the bug with printing the usage statement is not the source of your issue.

When you call run_metplus.py, the parm/metplus_config/defaults.conf file is read first automatically. It does not need to be passed into the script as an argument. Each argument passed to the run script will be read to set configuration variables defined and override the values of any variables that have been previously read. The defaults.conf file contains this line:

`PROCESS_LIST = Usage`

which will call the UsageWrapper by default if you don't pass in any arguments to the run script that override the PROCESS_LIST. The Example.conf file contains this line:

`PROCESS_LIST = Example`

which will call the ExampleWrapper. However, since you have passed in the defaults.conf after the Example.conf file, the value for PROCESS_LIST will be changed back to:

`PROCESS_LIST = Usage`

Side note: you no longer need to provide the "-c" flag in front of any command line arguments to run_metplus.py, but it does not hurt to leave them in there.

Instead of passing in the defaults.conf file on the command line, I recommend that you create a configuration file that is specific to you and the machine you are running on, override any settings that you want, then pass in that file at the end of your call to the run script.

More information on what must be changed from the default values can be found here: https://metplus.readthedocs.io/en/main_v4.0/Users_Guide/systemconfiguration.html#default-configuration-file

A information on how to set up a user configuration file can be found here:
https://metplus.readthedocs.io/en/main_v4.0/Users_Guide/systemconfiguration.html#user-configuration-file
